test/logflare_web/live_views/alerts_live_test.exs

```diff
@@ -17,6 +17,7 @@ defmodule LogflareWeb.AlertsLiveTest do
     insert(:plan, name: "Free")
     user = insert(:user)
     conn = login_user(conn, user)
+    start_supervised!(Logflare.Alerting.Supervisor)
```

Comment:
Starting the global process in multiple test suites prevents them from running `async: true`. Currently they are not async anyway, but that's something you generally want to do as much as possible and designing for it is good.

I see you have a separate application supervision tree for test env and you don't include `Logflare.Alerting.Supervisor`. Is there a reason not to include it though? It's generally best to keep the application supervision tree as close as possible in test and prod.

Reply:
we could rely on the macro to do it of course... as we do for some of the caches already
https://github.com/Logflare/logflare/commit/35e9cb38c4ffad3c65d1bd14772aa10c434943eb
not fully sold on the benefits of adding it to the test tree though 😄

Reply:
> not fully sold on the benefits of adding it to the test tree though 😄

Fair, ultimately it's up to you! The point I would stress is inability to `async: true`.

> would prefer to not start the alerting supervision tree during tests as the scheduler is stateful and the teardown of tests becomes quite important to ensure state is reset between tests

Statefulness is not necessarily a problem, it just needs to be accounted for in assertions. For example, instead of strict comparisons `list_alert_jobs() == [...]`, you would assert facts related to the given test, such as `new_job in list_alert_jobs()`.